lib/ansible/galaxy/role.py

```diff
@@ -400,7 +415,32 @@ def install(self):
                                     n_attr_value = to_native(attr_value)
                                     n_archive_parent_dir = to_native(archive_parent_dir)
                                     n_parts = n_attr_value.replace(n_archive_parent_dir, "", 1).split(os.sep)
+
+                                    if attr == 'name' and not n_attr_value.startswith(n_archive_parent_dir):
+                                        # Doesn't happen with our build process, but avoid extracting members that aren't adjacent to the role metadata
+                                        display.warning(f"Only extracting members in {n_archive_parent_dir}: ignoring {member.name}")
+                                        ignore_external.add(member)
+                                        continue
```

Comment:
If a tarfile contains something like:
```
./role-data/
├── role-name
│   ├── meta/main.yml
│   └── tasks
└── hitchhiker
    └── keep
```
we find the meta/main.yml in the role-name directory, but install all tarfile members (not just those adjacent to the meta directory). On devel this ends up installing an invalid role (and so it needs to be manually deleted to reinstall).

This could be an error instead, but I thought the warning was better at least than installing a broken role.

Should I separate this from the other changes or is it something we want to backport with the symlinks fix?